packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
 import 'package:meta/meta.dart';
 
-import 'print.dart';
+/// Styles for displaying a node in a [DiagnosticsNode] tree.
+///
+/// See also:
+///
+///  * [DiagnosticsNode.toStringDeep], which dumps text art trees for these
+///    styles.
+///  * [_normalTextConfiguration], which shows what text art for the standard
+///    style looks like.
+enum DiagnosticsTreeStyle {
+
+  /// Normal style for displaying trees.
+  normal,
+
+  /// Style that indicates a node is offstage.
+  offstage,
+
+  /// Slightly more compact version of the normal style.
+  ///
+  /// See also:
+  ///  * [Element], which uses this style.
+  dense,
+
+  /// Surround a node in the tree with a box.
+  ///
+  /// See also:
+  ///
+  ///  * [RenderParagraph], which uses this style.
+  box,
+
+  /// Render the tree just using whitespace without connecting parents to
+  /// children using lines.
+  ///
+  /// See also:
+  ///  * [SliverGeometry], which uses this style.
+  whitespace,
+
+  /// Render the tree on a single line without showing children.
+  singleLine,
+}
```

Comment:
Some of these names ("box" in particular, but also "whitespace", "singleLine", and to a lesser extent "dense") should be more semantic, rather than presentational, so that they can be more logically understood in the context of e.g. a visual debugger.